<launch>
  <arg name="input/lanelet_map_bin" default="/map/vector_map" />
  <arg name="input/obstacle_pointcloud" default="/sensing/lidar/no_ground/pointcloud" />
  <arg name="input/reference_trajectory" default="/planning/scenario_planning/trajectory" />
  <arg name="input/predicted_trajectory" default="/control/trajectory_follower/predicted_trajectory" />
  <arg name="input/twist" default="/localization/twist" />
  <arg name="config_file" default="$(find obstacle_collision_checker)/config/obstacle_collision_checker.yaml" />

  <node pkg="obstacle_collision_checker" type="obstacle_collision_checker_node" name="obstacle_collision_checker_node" output="screen">
    <remap from="~input/lanelet_map_bin" to="$(arg input/lanelet_map_bin)"/>
    <remap from="~input/obstacle_pointcloud" to="$(arg input/obstacle_pointcloud)"/>
    <remap from="~input/reference_trajectory" to="$(arg input/reference_trajectory)"/>
    <remap from="~input/predicted_trajectory" to="$(arg input/predicted_trajectory)"/>
    <remap from="~input/twist" to="$(arg input/twist)"/>

    <rosparam command="load" file="$(arg config_file)" />
  </node>
</launch>
